### PR TITLE
Twitter integration (v0.1)

### PR DESCRIPTION
**Pull request by makalaaneesh**
@dvargas92495  
This is for https://github.com/dvargas92495/roam-js-extensions/issues/64

Not sure how/if I can test the AWS lambda locally! So, pls do help with testing!
I can DM a twitter bearer token if you need.

### CODE REVIEW

File: src/lambda-helpers.ts
Comment (dvargas92495) [inline src/lambda-helpers.ts:44]: we're going to want to add this environment variable to the github action and I'll need to add the secret to the repo:

https://github.com/dvargas92495/roam-js-extensions/blob/master/.github/workflows/master_lambda.yaml#L8-L13

File: src/entries/twitter.ts
Comment (dvargas92495) [inline src/entries/twitter.ts:1]: Let's add a doc on https://github.com/dvargas92495/roam-js-extensions/tree/master/docs

**Review by dvargas92495 — COMMENTED** [review]
lambdas only deploy from master, which has led me to create this: https://github.com/dvargas92495/roam-js-extensions/issues/69. since it looks straightforward, I'm cool merging to master to unlock testing.

Before merging, I'd like to iterate on the period PR first and rebase this branch with those changes. left some other feedback on this PR!

File: src/entries/twitter.ts
Comment (makalaaneesh) [reply src/entries/twitter.ts:16]: I actually had just copy pasted without knowing what it does 😅  
I don't have any use for it. I'll just remove it !

File: src/lambda-helpers.ts
Comment (makalaaneesh) [reply src/lambda-helpers.ts:44]: done

File: src/entries/twitter.ts
Comment (makalaaneesh) [reply src/entries/twitter.ts]: done

**Review by dvargas92495 — COMMENTED** [review]
Cool to approve once we rebase and add doc :+1:

File: src/lambda-helpers.ts
Comment (dvargas92495) [reply src/lambda-helpers.ts:44]: looking at the docs, it looks like we have to generate a bearer token on each request based on a consumer key and secret? https://developer.twitter.com/en/docs/authentication/oauth-2-0

Comment (makalaaneesh) [reply src/lambda-helpers.ts:44]: Thanks for this comment :) 

I couldn't find an explicit mention that they might expire and we'd have to regenerate them. This thought crossed my mind too, but this [doc](https://developer.twitter.com/en/docs/authentication/oauth-2-0/bearer-tokens) says `You can also obtain a Bearer Token from the developer portal inside the keys and tokens section of your App's settings.`, so I assumed they were static. 

I'll read up more on oauth bearer tokens and get back to you.

Comment (makalaaneesh) [reply src/lambda-helpers.ts]: https://developer.twitter.com/en/docs/authentication/faq states that access tokens are not explicitly expired. So, while that shouldn't be a problem, I think it's a better approach to store the consumer Key and Secret and get the access token before making a request. That way even if we have to invalidate the bearer token, we need not make changes to the application.

I've pushed a commit with the above changes. 

Noting that since I can't really test the lambda locally + I can't call twitter API directly because of the CORS restrictions, the lambda code is essentially untested!

**Comment by makalaaneesh** [comment]
@dvargas92495 
- rebased
- added docs (video is pending. Will do it after testing with master)

File: .github/workflows/master_lambda.yaml
Comment (dvargas92495) [inline .github/workflows/master_lambda.yaml:15]: Should probably call this `TWITTER_CONSUMER_KEY` right?

EDIT: yea we definitely need this to match https://github.com/dvargas92495/roam-js-extensions/pull/66/files#diff-5d4f96efb0abeb3fa4cc55154d4f9307R47

File: docs/extensions/twitter.md
Comment (dvargas92495) [inline docs/extensions/twitter.md:3]: could we use a less polarizing example lol. don't actually care too much

File: src/lambda-helpers.ts
Comment (dvargas92495) [inline src/lambda-helpers.ts:66]: this is really dope, great work! going to go add the twitter secrets now

**Review by dvargas92495 — COMMENTED** [review]
Just 1 small edit in the github actions file, then I'm ready to approve and merge! Amazing stuff!

File: .github/workflows/master_lambda.yaml
Comment (makalaaneesh) [reply .github/workflows/master_lambda.yaml]: My bad, fixed!

File: docs/extensions/twitter.md
Comment (makalaaneesh) [reply docs/extensions/twitter.md]: Haha. Changed it :)

File: src/lambda-helpers.ts
Comment (makalaaneesh) [reply src/lambda-helpers.ts:66]: Ah, thanks! You were my inspiration!

**Review by dvargas92495 — APPROVED** [review]
Let's test it!

**Pull request merged**
